docs/units/distributions.rst

```diff
@@ -0,0 +1,243 @@
+.. |quantity| replace:: :class:`~astropy.units.Quantity`
+.. |distribution| replace:: :class:`~astropy.units.Distribution`
+
+.. _unit_distributions:
+
+Distributions
+*************
+
+In addition to |quantity|, astropy provides a |distribution| object to represent
+statistical distributions in a form that acts as a drop-in replacement for
+|quantity| or regulay Numpy arrays. USed in this manner, |distribution| provides
+uncertainty propogation at the cost of additional computation.  It can also
+more generally represent sampled distributions for e.g., Monte Carlo calculation
+techniques.
+
+The core object for this feature is the |distribution|.  Currently, all
+such distributions are Monte Carlo sampled.  While this means each distribution
+may take more memory, it allows arbitrarily complex operations to be performed
+on distributions while maintaining their correlation structure. Some specific
+well-behaved distributions (e.g., `~astropy.units.NormalDistribution`) have
+analytic forms which may eventually enable a more compact/efficient
+representation, but this is not currently implemented.
+
+
+Creating distributions
+======================
+
+The most conceptually straightforward way to create a distribution is to use an
+array or quantity that carries the samples in the *last* dimension:

```

Comment:
I feel like it would actually be better to first show the more common case of the well defined distributions and *then* to show how do create arbitrary distributions. Creating the more common ones is easier since it doesn't rely on having to make samples manually. The first example could use only e.g. gaussians and show a simple calculation with a real example (e.g. take an equation such as 1/2 M v^2 and then show how to do uncertainty propagation.

Reply:
I see your point here... and that might be more natural with the classmethod/function way